### PR TITLE
[CDF-21821, CDF-21860, CDF-22355] Split auth in inti and verify

### PR DESCRIPTION
**Pull request by doctrino**
# Description

## Init
image

## Verify
More or less the same as the old `auth verify`, but based on the required loaders capabilities and not the admin group.  Thus, a simpler interface of only one flag `--dry-run`

### Listing Resources that requires the capabilities

image

### Merging Capabilities
image

### Fix
The previous `cdf auth verify` could end up removing existing capabilities, the new version ensures it is only adding.

### Note
As the commands are interactive the testing is outdated. I have done manually testing and set the existing tests to skip. Suggest we move this to the backlog as rewriting the tests to work with interactive is a significant effort. In addition, there is more refactoring that can be done as interactive is now an assumption, but trying to limit the scope of this PR to not do more changes than necessary.

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/toolkit/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/toolkit/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/_cdf_tk/apps/_auth_app.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        ctx: typer.Context,
+        dry_run: Annotated[
+            bool,
+            typer.Option(
+                "--dry-run",
+                "-r",
+                help="Whether to do a dry-run. This means that no changes to CDF will be made.",
+            ),
+        ] = False,
+        no_prompt: Annotated[
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/apps/_auth_app.py:68]: Alternative name `--no-interactive`. Pro; Maybe more descriptive, Con: longer

File: cognite_toolkit/_cdf_tk/apps/_auth_app.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        dry_run: Annotated[
+            bool,
+            typer.Option(
+                "--dry-run",
+                "-r",
+                help="If you verify, and you pass this flag no changes to CDF will be made.",
+            ),
+        ] = False,
+    ) -> None:
+        """Creates the authorization for a user/service principal to run the CDF Toolkit commands.
```
Comment (ronpal) [inline cognite_toolkit/_cdf_tk/apps/_auth_app.py:41]:
```suggestion
        """Sets the OIDC parameters required to authenticate and authorize the Cognite Toolkit in Cognite Data Fusion.
```

File: cognite_toolkit/_cdf_tk/apps/_auth_app.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                help="Whether to do a dry-run. This means that no changes to CDF will be made.",
+            ),
+        ] = False,
+        no_prompt: Annotated[
+            bool,
+            typer.Option(
+                "--no-prompt",
+                "-np",
+                help="Whether to skip the prompt to continue. This is useful for CI/CD pipelines."
+                "If you passe this flag, the command will exit with 1 if the user/service principal does not have the required capabilities.",
```
Comment (ronpal) [inline cognite_toolkit/_cdf_tk/apps/_auth_app.py:74]:
```suggestion
                "If you include this flag, the execution will stop if the user or service principal does not have the required capabilities.",
```

File: cognite_toolkit/_cdf_tk/apps/_auth_app.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            ),
+        ] = False,
+    ) -> None:
+        """Verify that the current user/service principal has the required capabilities to run the CDF Toolkit commands."""
```
Comment (ronpal) [inline cognite_toolkit/_cdf_tk/apps/_auth_app.py:78]:
```suggestion
        """Verify that the current user or service principal has the required capabilities to run the CDF Toolkit commands."""
```

File: CHANGELOG.cdf-tk.md [hunk 5]
```diff
@@ -17,6 +17,15 @@ Changes are grouped as follows:
 
 ## TBD
 
+### Changed
+
+- [BREAKING] The command `cdf auth verify` has been split into `cdf auth init` and `cdf auth verify`. The `init` command
+  is used to initialize the authentication, and the `verify` command is used to verify the authentication. The `init`
```
Comment (ronpal) [inline CHANGELOG.cdf-tk.md:23]:
```suggestion
  is used to initialize the auth parameters, and the `verify` command is used to verify that required privileges are set. The `init`
```

File: cognite_toolkit/_cdf_tk/commands/auth.py [hunk 6]
```diff
@@ -132,39 +116,51 @@ def check_auth(
...
-            if interactive:
-                to_create, to_delete = self.upsert_toolkit_group_interactive(principal_groups, admin_write_group)
-            else:
-                to_create, to_delete = self.upsert_toolkit_group(
-                    principal_groups, admin_write_group, update_group, create_group
-                )
+        if missing_capabilities and not is_interactive:
+            raise AuthorizationError(
+                "The service principal/application does not have the required capabilities to run the Toolkit with "
+                "all resources supported by Toolkit"
```
Comment (ronpal) [inline cognite_toolkit/_cdf_tk/commands/auth.py:137]:
```suggestion
                "The service principal/application does not have the required capabilities for the Toolkit to function properly"
```

**Review by ronpal — APPROVED** [review]

**Pull request merged**